### ISSUE TITLE
crash in test_sycl_kernel_submit.py

### ISSUE THREAD

**Issue by halcyon-gh** [IntelPython/dpctl#1388]
In figuring out why my local install of dpctl was not working I tried running the included tests. This results in a consistent crash in the test_sycl_kernel_submit.py file instead of a failed test. The conda environment consisted of python 3.10.12 and dpctl 0.14.5 + dependencies from the intel channel. If you need more details, please let me know.

I assume that whatever the issue is, the test case should report an error rather than crash. Next to that I hope you might have some suggestions what could be causing the problem in my setup. 

```
(dpctl2) C:\-\dpctl\tests>pytest test_sycl_kernel_submit.py
=== test session starts ===
platform win32 -- Python 3.10.12, pytest-7.4.0, pluggy-1.0.0
rootdir: C:\-\dpctl\tests
collected 8 items

test_sycl_kernel_submit.py
(dpctl2) C:\-\dpctl\tests>
```

**Comment by oleksandr-pavlyk** [comment]
@halcyon-gh Please include output of `conda list --explicit`, as well as the output of `python -m dpctl -f`.

**Comment by halcyon-gh** [comment]
```
(dpctl2) C:\-\>conda list --explicit 
# This file may be used to create an environment using:
# $ conda create --name <env> --file <this file>
# platform: win-64
@EXPLICIT
https://conda.anaconda.org/intel/win-64/ca-certificates-2023.5.7-h56e8100_0.tar.bz2
https://conda.anaconda.org/intel/win-64/intel-cmplr-lib-rt-2023.2.0-intel_49496.tar.bz2
https://conda.anaconda.org/intel/win-64/intel-cmplr-lic-rt-2023.2.0-intel_49496.tar.bz2
https://conda.anaconda.org/intel/win-64/intel-openmp-2023.2.0-intel_49496.tar.bz2
https://conda.anaconda.org/intel/win-64/intelpython-2023.2.0-0.tar.bz2
https://repo.anaconda.com/pkgs/main/noarch/tzdata-2023c-h04d1e81_0.conda
https://conda.anaconda.org/intel/win-64/ucrt-10.0.22621.0-h57928b3_0.tar.bz2
https://conda.anaconda.org/intel/win-64/vc14_runtime-14.34.31931-h5081d32_16.tar.bz2
https://conda.anaconda.org/intel/win-64/vs2015_runtime-14.34.31931-hed1258a_16.tar.bz2
https://repo.anaconda.com/pkgs/main/win-64/vc-14.2-h21ff451_1.conda
https://repo.anaconda.com/pkgs/main/win-64/bzip2-1.0.8-he774522_0.conda
https://repo.anaconda.com/pkgs/main/win-64/cfitsio-3.470-h2bbff1b_7.conda
https://repo.anaconda.com/pkgs/main/win-64/charls-2.2.0-h6c2663c_0.conda
https://repo.anaconda.com/pkgs/main/win-64/giflib-5.2.1-h8cc25b3_3.conda
https://repo.anaconda.com/pkgs/main/win-64/jpeg-9e-h2bbff1b_1.conda
https://repo.anaconda.com/pkgs/main/win-64/jxrlib-1.1-he774522_2.conda
https://repo.anaconda.com/pkgs/main/win-64/lerc-3.0-hd77b12b_0.conda
https://repo.anaconda.com/pkgs/main/win-64/libaec-1.0.4-h33f27b4_1.conda
https://repo.anaconda.com/pkgs/main/win-64/libbrotlicommon-1.0.9-h2bbff1b_7.conda
https://repo.anaconda.com/pkgs/main/win-64/libdeflate-1.17-h2bbff1b_0.conda
https://repo.anaconda.com/pkgs/main/win-64/libffi-3.4.4-hd77b12b_0.conda
https://repo.anaconda.com/pkgs/main/win-64/libwebp-base-1.2.4-h2bbff1b_1.conda
https://repo.anaconda.com/pkgs/main/win-64/libzopfli-1.0.3-ha925a31_0.conda
https://repo.anaconda.com/pkgs/main/win-64/lz4-c-1.9.4-h2bbff1b_0.conda
https://conda.anaconda.org/intel/win-64/openssl-3.1.1-hcfcfb64_1.tar.bz2
https://repo.anaconda.com/pkgs/main/win-64/snappy-1.1.9-h6c2663c_0.conda
https://repo.anaconda.com/pkgs/main/win-64/sqlite-3.41.2-h2bbff1b_0.conda
https://conda.anaconda.org/intel/win-64/tbb-2021.10.0-vc14_intel_49542.tar.bz2
https://repo.anaconda.com/pkgs/main/win-64/tk-8.6.12-h2bbff1b_0.conda
https://repo.anaconda.com/pkgs/main/win-64/xz-5.4.2-h8cc25b3_0.conda
https://repo.anaconda.com/pkgs/main/win-64/yaml-0.2.5-he774522_0.conda
https://repo.anaconda.com/pkgs/main/win-64/zfp-0.5.5-hd77b12b_6.conda
https://repo.anaconda.com/pkgs/main/win-64/zlib-1.2.13-h8cc25b3_0.conda
https://conda.anaconda.org/intel/win-64/intel-opencl-rt-2023.2.0-intel_49496.tar.bz2
https://repo.anaconda.com/pkgs/main/win-64/libbrotlidec-1.0.9-h2bbff1b_7.conda
https://repo.anaconda.com/pkgs/main/win-64/libbrotlienc-1.0.9-h2bbff1b_7.conda
https://repo.anaconda.com/pkgs/main/win-64/libpng-1.6.39-h8cc25b3_0.conda
https://conda.anaconda.org/intel/win-64/mkl-2023.2.0-intel_49496.tar.bz2
https://repo.anaconda.com/pkgs/main/win-64/python-3.10.12-he1021f5_0.conda
https://repo.anaconda.com/pkgs/main/win-64/zstd-1.5.5-hd43e919_0.conda
https://conda.anaconda.org/conda-forge/noarch/array-api-compat-1.3-pyhd8ed1ab_0.conda
https://repo.anaconda.com/pkgs/main/win-64/blosc-1.21.3-h6c2663c_0.conda
https://repo.anaconda.com/pkgs/main/win-64/brotli-bin-1.0.9-h2bbff1b_7.conda
https://repo.anaconda.com/pkgs/main/win-64/colorama-0.4.6-py310haa95532_0.conda
https://repo.anaconda.com/pkgs/main/win-64/exceptiongroup-1.0.4-py310haa95532_0.conda
https://repo.anaconda.com/pkgs/main/noarch/iniconfig-1.1.1-pyhd3eb1b0_0.tar.bz2
https://repo.anaconda.com/pkgs/main/win-64/libtiff-4.5.1-hd77b12b_0.conda
https://repo.anaconda.com/pkgs/main/win-64/packaging-23.1-py310haa95532_0.conda
https://repo.anaconda.com/pkgs/main/win-64/pluggy-1.0.0-py310haa95532_1.conda
https://repo.anaconda.com/pkgs/main/win-64/pyyaml-6.0-py310h2bbff1b_1.conda
https://repo.anaconda.com/pkgs/main/win-64/setuptools-68.0.0-py310haa95532_0.conda
https://conda.anaconda.org/intel/noarch/six-1.16.0-pyhd3eb1b0_1.tar.bz2
https://conda.anaconda.org/intel/win-64/tbb4py-2021.10.0-py310_intel_49542.tar.bz2
https://repo.anaconda.com/pkgs/main/win-64/tomli-2.0.1-py310haa95532_0.conda
https://repo.anaconda.com/pkgs/main/win-64/wheel-0.38.4-py310haa95532_0.conda
https://repo.anaconda.com/pkgs/main/win-64/brotli-1.0.9-h2bbff1b_7.conda
https://repo.anaconda.com/pkgs/main/win-64/lcms2-2.12-h83e58a3_0.conda
https://conda.anaconda.org/intel/win-64/mkl-service-2.4.0-py310h71267b0_15.tar.bz2
https://repo.anaconda.com/pkgs/main/win-64/openjpeg-2.4.0-h4fc8c34_0.conda
https://repo.anaconda.com/pkgs/main/win-64/pip-23.2.1-py310haa95532_0.conda
https://repo.anaconda.com/pkgs/main/win-64/pytest-7.4.0-py310haa95532_0.conda
https://repo.anaconda.com/pkgs/main/win-64/imagecodecs-2021.8.26-py310h4c966c4_2.conda
https://repo.anaconda.com/pkgs/main/noarch/tifffile-2021.7.2-pyhd3eb1b0_2.conda
https://conda.anaconda.org/intel/win-64/dpcpp-cpp-rt-2023.2.0-intel_49496.tar.bz2
https://conda.anaconda.org/intel/win-64/dpcpp_cpp_rt-2023.2.0-intel_49496.tar.bz2
https://conda.anaconda.org/intel/win-64/icc_rt-2023.2.0-intel_49496.tar.bz2
https://conda.anaconda.org/intel/win-64/numpy-base-1.23.5-py310he7338a5_7.tar.bz2
https://conda.anaconda.org/intel/win-64/mkl_fft-1.3.5-py310h11997cf_11.tar.bz2
https://conda.anaconda.org/intel/win-64/mkl_random-1.2.2-py310hb8cdf03_51.tar.bz2
https://conda.anaconda.org/intel/win-64/mkl_umath-0.1.1-py310h7cd8e37_61.tar.bz2
https://conda.anaconda.org/intel/win-64/numpy-1.23.5-py310hb1a71d1_7.tar.bz2
https://conda.anaconda.org/intel/win-64/dpctl-0.14.5-py310hdcf5b23_24.tar.bz2
```

```
(dpctl2) C:\-\>python -m dpctl -f
Platform  0 ::
    Name        Intel(R) OpenCL
    Version     OpenCL 3.0 WINDOWS
    Vendor      Intel(R) Corporation
    Backend     opencl
    Num Devices 1
      # 0
        Name                Intel(R) Core(TM) i9-10885H CPU @ 2.40GHz
        Version             2023.16.6.0.28_042959
        Filter string       opencl:cpu:0
Platform  1 ::
    Name        Intel(R) OpenCL HD Graphics
    Version     OpenCL 3.0
    Vendor      Intel(R) Corporation
    Backend     opencl
    Num Devices 1
      # 0
        Name                Intel(R) UHD Graphics
        Version             27.20.100.9168
        Filter string       opencl:gpu:0
Platform  2 ::
    Name        Intel(R) OpenCL
    Version     OpenCL 3.0 WINDOWS
    Vendor      Intel(R) Corporation
    Backend     opencl
    Num Devices 1
      # 0
        Name                Intel(R) Core(TM) i9-10885H CPU @ 2.40GHz
        Version             2022.15.12.0.01_081451
        Filter string       opencl:cpu:1
Platform  3 ::
    Name        Intel(R) FPGA Emulation Platform for OpenCL(TM)
    Version     OpenCL 1.2 Intel(R) FPGA SDK for OpenCL(TM), Version 20.3
    Vendor      Intel(R) Corporation
    Backend     opencl
    Num Devices 1
      # 0
        Name                Intel(R) FPGA Emulation Device
        Version             2022.15.12.0.01_081451
        Filter string       opencl:accelerator:0
```

**Comment by oleksandr-pavlyk** [comment]
I have recreated your environment using the provided output of the ``conda list --explicit``. 

Here is my configuration:

```
(gh_1388) C:\Users\opavlyk>python -m dpctl -f
Platform  0 ::
    Name        Intel(R) OpenCL
    Version     OpenCL 3.0 WINDOWS
    Vendor      Intel(R) Corporation
    Backend     opencl
    Num Devices 1
      # 0
        Name                11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
        Version             2023.16.6.0.28_042959
        Filter string       opencl:cpu:0
Platform  1 ::
    Name        Intel(R) OpenCL HD Graphics
    Version     OpenCL 3.0
    Vendor      Intel(R) Corporation
    Backend     opencl
    Num Devices 1
      # 0
        Name                Intel(R) Iris(R) Xe Graphics
        Version             31.0.101.4032
        Filter string       opencl:gpu:0
Platform  2 ::
    Name        Intel(R) Level-Zero
    Version     1.3
    Vendor      Intel(R) Corporation
    Backend     ext_oneapi_level_zero
    Num Devices 1
      # 0
        Name                Intel(R) Iris(R) Xe Graphics
        Version             1.3.24931
        Filter string       level_zero:gpu:0
```

I am able to run the test just fine:

```
(gh_1388) C:\Users\opavlyk>python -m pytest --pyargs dpctl.tests.test_sycl_kernel_submit
================================================= test session starts =================================================
platform win32 -- Python 3.10.12, pytest-7.4.0, pluggy-1.0.0
rootdir: C:\Users\opavlyk
collected 8 items

Miniconda3\envs\gh_1388\lib\site-packages\dpctl\tests\test_sycl_kernel_submit.py ......s.                        [100%]

============================================ 7 passed, 1 skipped in 18.77s ============================================
```

One thing that stands out is that your driver is very much older than mine "27.20.100.9168" vs "31.0.101.4032". 

This makes me suggest to prompt you to update the driver first. To do that, open "Device Manager", select "Display Adapters", right click on UHD device entry, select Properties, and navigate to the Driver tab. You could try using "Update driver" button on the interface. 

Alternatively you can get drivers from https://www.intel.com/content/www/us/en/support/products/80939/graphics.html

If your compute has Gen9 GPU hardware (as it seems per [Ark](https://ark.intel.com/content/www/us/en/ark/products/203682/intel-core-i910885h-processor-16m-cache-up-to-5-30-ghz.html) spec of your CPU), the latest driver version is " 31.0.101.2121" and for it the test is known to pass.

**Comment by halcyon-gh** [comment]
Thanks for the suggestion. This seems to have fixed the issue after some initial weirdness. Feel free to close the issue.

After updating drivers and rebooting, I ran the tests again and got some 'access error' message. I then tried running the tests with specific devices using SYCL_DEVICE_FILTER, which hard locked windows for the `opencl:gpu` selection. After rebooting again all test run successfully and I cannot reproduce the errors I got before.
